imap/jmap_contact.c

```diff
@@ -1679,15 +1698,7 @@ static json_t *jmap_contact_from_vcard(const char *userid,
             /* Read countryCode from same-grouped ABADR property, if any */
             const char *countrycode = NULL;
             if (entry->group) {
-                // O(n^2) n is the (presumably small) count of VCARD props
-                struct vparse_entry *iter;
-                for (iter = card->properties; iter; iter = iter->next) {
-                    if (!strcasecmpsafe(iter->group, entry->group) &&
-                            !strcasecmp(iter->name, VCARD_APPLE_ABADR_PROPERTY)) {
-                        countrycode = iter->v.value;
-                        break;
-                    }
-                }
+                countrycode = hash_lookup(entry->group, &abadr_by_group);
```

Comment:
@rsto

The previous code would compare the group names case insensitively, the new code looks for an exact match.

Is that a problem? Does the space say ITEM1 and item1 are the same or different?